android/expoview/src/main/java/versioned/host/exp/exponent/modules/api/ContactsModule.java

```diff
@@ -61,116 +108,127 @@ public void getContactsAsync(final ReadableMap options, final Promise promise) {
       return;
     }
 
+    Set<String> fieldsSet = getFieldsSet(options.getArray("fields"));
+
     int pageOffset = options.getInt("pageOffset");
     int pageSize = options.getInt("pageSize");
-    Set<String> fieldsSet = getFieldsSet(options.getArray("fields"));
     boolean fetchSingleContact = options.hasKey("id");
-    WritableArray contacts = Arguments.createArray();
     WritableMap response = Arguments.createMap();
 
-    ContentResolver cr = getReactApplicationContext().getContentResolver();
-    Cursor cursor = cr.query(
-        ContactsContract.Contacts.CONTENT_URI,
-        null,
-        fetchSingleContact ? Contacts._ID + " = ?" : null,
-        fetchSingleContact ? new String[] { options.getString("id") } : null,
-        null
-    );
-    if (cursor != null) {
-      try {
-        cursor.move(pageOffset);
-        int currentIndex = 0;
-        Set<Long> ids = new HashSet<>();
+    Map<String, Contact> contacts;
 
-        final int contactIdIndex = cursor.getColumnIndex(Contacts._ID);
-        while (cursor.moveToNext()) {
-          if (currentIndex >= pageSize) {
-            break;
-          }
+    ContentResolver cr = getReactApplicationContext().getContentResolver();
+    Cursor cursor;
+    if (fetchSingleContact)
+      cursor = cr.query(
+          ContactsContract.Data.CONTENT_URI,
+          FULL_PROJECTION.toArray(new String[FULL_PROJECTION.size()]),
+          ContactsContract.Data.CONTACT_ID + " = ?",
+          new String[] { options.getString("id") },
+          null
+      );
+    else {
+
+
+      String selection = ContactsContract.Data.MIMETYPE + "=? OR " + ContactsContract.Data.MIMETYPE + "=? OR " + ContactsContract.Data.MIMETYPE + "=? OR " + ContactsContract.Data.MIMETYPE + "=? OR " + ContactsContract.Data.MIMETYPE + "=?";
```

Comment:
Does this work as expected? Eg. `prefix` is not added to selection but is still there in the response. If this must be there - please give this some formatting - every "or" in single line :)

Reply:
check